htdocs/societe/soc.php

```diff
@@ -2335,8 +2335,13 @@
             $result=$adh->fetch('','',$object->id);
             if ($result > 0)
             {
+            	if ($adh->morphy=='mor'){
+            		$adh->ref=$adh->company;
```

Comment:
Because the identity of a member is the name (so when it is a moral person, company name is often store also into name), i suggest this instead
$adh->ref=$adh->getFullName($langs);